recipes/boost/all/conanfile.py

```diff
@@ -478,6 +486,8 @@ def _run_python_script(self, script):
         # Conan is broken when run_to_output = True
         if "\n-----------------\n" in output:
             output = output.split("\n-----------------\n", 1)[1]
+        elif re.search('\n----+$', output) != None: 
```

Comment:
I am curious, why above two lines didn't strip bad lines for you?
I tried locally, and for me it works correctly on Windows
but seems like in your case, it's still returning an output with dashes not removed...
maybe you can dump an unmodified output before line 486?

Reply:
/cc @madebr as seems like he did an original workaround

Reply:
@SSE4 The output of the `self.run()` function ends with `\n-----------------\n<script output>` if the script does return output. But when the script returns nothing the output of the `self.run()` function ends with `\n-----------------` so the output is not split.
There are of course alternative implementations, we could also split at `\n-----------------` and then strip a possible leading `\n` from the result.

On my Windows machine the script that _python_abiflags(self) runs gives no (empty) output, that's were the build messed up.

Reply:
oh, I see now, and can reproduce: the `strip()` call removes `\n` at the end, so output doesn't match `\n-----------------\n` any longer.
the following fixes abi flags for me:
```
index 359e4a3..98caa84 100644
--- a/recipes/boost/all/conanfile.py
+++ b/recipes/boost/all/conanfile.py
@@ -501,10 +501,12 @@ class BoostConan(ConanFile):
         except ConanException:
             self.output.info("(failed)")
             return None
-        output = output.getvalue().strip()
+        output = output.getvalue()
         # Conan is broken when run_to_output = True
         if "\n-----------------\n" in output:
             output = output.split("\n-----------------\n", 1)[1]
+        output = output.strip()
```

Reply:
@SSE4 's implementation looks easier to understand.
Let's hope switching to conan 2.0 will drop the need for this workaround.

Reply:
Confirm, that works also for me. I changed the code accordingly.